modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -54,39 +62,46 @@ extension AddressBookClient: DependencyKey {
                     let encryptedFileURL = documentsDirectory.appendingPathComponent(try AddressBookClient.filenameForEncryptedFile())
 
                     if let contactsData = try? Data(contentsOf: encryptedFileURL) {
-                        // file exists, try to find the unencrypted file in case the delete failed
+                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
+                        
+                        // file exists and was successfully decrypted and parsed;
+                        // try to find the unencrypted file and delete it
                         let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
                         if FileManager.default.fileExists(atPath: unencryptedFileURL.path) {
                             try? FileManager.default.removeItem(at: unencryptedFileURL)
                         }
-                        
-                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
 
                         latestKnownContacts = contacts
-                        return contacts
+                        return (contacts, nil)
                     } else {
                         // Fallback to the unencrypted file check and resolution
                         let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
                         
                         if let contactsData = try? Data(contentsOf: unencryptedFileURL) {
-                            // file exists, ensure data are parsed, re-saved and file deteled
-                            let contacts = try AddressBookClient.contactsFrom(unencryptedData: contactsData)
+                            // Unencrypted file exists; ensure data are parsed, re-saved as encrypted, and the original file deleted.
+
+                            var contacts = try AddressBookClient.contactsFrom(contactsData)
 
                             // try to encrypt and store the data
+                            var remoteStoreResult: RemoteStoreResult
                             do {
-                                try AddressBookClient.storeContacts(contacts, remoteStorage: remoteStorage, remoteStore: false)
+                                remoteStoreResult = try AddressBookClient.storeContacts(contacts, remoteStorage: remoteStorage, remoteStore: false)
+
```

Comment:
I agree it's redundant in some scenarios but syncContacts is done in a way that it could throw an error and terminate the store completely. The current code ensure there is at least local store done and sync "can" fail. We might want to improve the order and robustness of the solution but I put it to follow up work in #1408

Reply:
If we want `syncContacts` to always store locally, there's an easy way to do that: set `cannotUpdateRemote = true` in those error cases (i.e. treat them the same way as `RemoteStorageError.containerURL`), rather than propagating the error directly. If we want to record the reason then that can be done as well. Anyway this has been merged as-is now, but please consider that approach for #1408.